### PR TITLE
Docs: tag based deployment

### LINKED ISSUE #289: Update documentation to reflect new deployment model
At a minimum, let's update
 - [x] [release documentation](https://docs.calitp.org/eligibility-server/releases/)
 - [x] [infrastructure documentation](https://docs.calitp.org/eligibility-server/deployment/infrastructure/)

Feel free to add to the list above with other documentation that I might've missed.

~Also, maybe we should add that infrastructure documentation to the docs site?~ done in https://github.com/cal-itp/eligibility-server/issues/293

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #289

### CODE REVIEW

**Comment by angela-tran** [comment]
This PR will be marked as ready after we do #290

**Comment by thekaveman** [comment]
Rebased on the latest `main`

File: docs/releases.md [hunk 1]
```diff
@@ -2,14 +2,26 @@
 
 The `eligibility-server` is published as a Docker image on the GitHub Container Registry. It can be accessed from the [repository package page](https://github.com/cal-itp/eligibility-server/pkgs/container/eligibility-server).
 
-Every push to the `main` (default) branch that changes files relevant to the application builds and updates the `dev` package, via the [`docker-publish`](https://github.com/cal-itp/eligibility-server/blob/main/.github/workflows/docker-publish.yml) GitHub Action.
+Every push to the `main` (default) branch that changes files relevant to the application builds and updates the `dev` package, via the [`docker-publish`](https://github.com/cal-itp/eligibility-server/blob/dev/.github/workflows/docker-publish.yml) GitHub Action.
```
Comment (angela-tran) [inline docs/releases.md:5]:
```suggestion
Every push to the `main` (default) branch that changes files relevant to the application builds and updates the `dev` package, via the [`docker-publish`](https://github.com/cal-itp/eligibility-server/blob/main/.github/workflows/docker-publish.yml) GitHub Action.
```

Comment (thekaveman) [reply docs/releases.md]: Thank you @angela-tran! I swear I did a search for `/dev` and thought I replaced them all :thinking:

**Review by thekaveman — APPROVED** [review]

**Pull request merged**